src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -170,8 +170,10 @@ final case class PurposeApiServiceImpl(
           Future.failed(MissingFreeOfChargeReason)
         else Future.unit
       purpose        <- purposeManagementService.getPurposeById(purposeUUID)
```

Comment:
we need to verify that an agreement exists between the requester (the consumer) and the new eserviceid.
This agreement should have a valid state (ACTIVE or SUSPENDED).
@beetlecrunch please confirm the allowed states